utils/build_rpms_from_scratch.sh

```diff
@@ -60,15 +134,62 @@ setup_rpm_build_env() {
     dnf config-manager --set-enabled powertools
   elif [[ "$el_version" == "9" ]]; then
     dnf config-manager --set-enabled crb # CodeReady Builder
-  else
-    echo "Unsupported EL Version: $el_version"
-    exit 1
   fi
 
   # ignore dpdk because want to use rpms build from daos-stack/dpdk
   dnf update -y --exclude=dpdk --exclude=dpdk-devel
 }
 
+l_createrepo() {
+  createrepo "$1"
+
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+    zypper refresh
+  fi
+}
+
+prepare_env() {
+  if [ -f /tmp/setup_done ]; then return; fi
+
+  rpms_dst=$1
+  build_dst=$2
+
+  mkdir -p "$rpms_dst" "$build_dst"
+
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+      prepare_env_leap
+    else
+      prepare_env_rocky
+  fi
+
+  setup_local_repo "$rpms_dst"
+  l_createrepo "$rpms_dst"
+
+  touch /tmp/setup_done
+}
+
+verify_os_version() {
+  $OS_ID
```

Comment:
I think this would be a syntax error here.  Maybe you meant:

```suggestion
  echo "$OS_ID"